src/scripts/main.js

```diff
@@ -1,3 +1,243 @@
+
+const updateCell = (cell, value = '') => {
+  cell.className = value ? `field-cell field-cell--${value}` : 'field-cell';
+  cell.textContent = value;
+};
+
+const moveHelper = (currentCell, nextCell, cellValue) => {
+  if (nextCell.classList.length > 1) {
+    return;
+  };
+
+  updateCell(currentCell);
+  updateCell(nextCell, cellValue);
+};
+
+const addHelper = (currentCell, nextCell, cellValue) => {
+  if (Number(nextCell.textContent) === Number(cellValue)) {
+    updateCell(currentCell);
+    updateCell(nextCell, cellValue * 2);
+    updateScore(cellValue * 2);
+
+    if (Number(cellValue * 2) === 2048) {
+      isPlaying = false;
+      messageWinElement.classList.remove('hidden');
+    }
+  };
+};
+
+const moveAddTiles = (tilesIndexes, direction, type = 'move') => {
+  let step;
+
+  switch (direction) {
+    case 'right':
+      step = 1;
+      break;
+    case 'left':
+      step = -1;
+      break;
+    case 'up':
+      step = -4;
+      break;
+    case 'down':
+      step = 4;
+      break;
+  };
+
+  tilesIndexes.forEach(index => {
+    const currentCell = cells[index];
+
+    if (currentCell.classList.length === 1) {
+      return;
+    };
+
+    const nextCell = cells[index + step];
+    const cellValue = currentCell.textContent;
+
+    if (type === 'move') {
+      moveHelper(currentCell, nextCell, cellValue);
+
+      return;
+    };
+    addHelper(currentCell, nextCell, cellValue);
+  });
+};
+
+const moveDirectionAndSum = (directionArr, direction) => {
+  moveAddTiles(directionArr, direction);
+  moveAddTiles(directionArr, direction);
+  moveAddTiles(directionArr, direction);
+  moveAddTiles(directionArr, direction, 'add');
+  moveAddTiles(directionArr, direction);
+  moveAddTiles(directionArr, direction);
+};
+
+const innit = () => {
+  appendNewNumberTile();
+  appendNewNumberTile();
+  isPlaying = true;
+};
+
+const startButtonCallback = () => {
+  innit();
+  startButton.remove();
+  restartButton.hidden = false;
+  messageStartElement.classList.add('hidden');
+};
+
+const restartButtonCallback = () => {
+  updateScore(0);
+  messageLoseElement.classList.add('hidden');
+  messageWinElement.classList.add('hidden');
+
+  for (const cell of cells) {
+    cell.className = 'field-cell';
+    cell.textContent = '';
+  };
+  innit();
+};
+
+const keydownCallback = (e) => {
+  e.preventDefault();
+
+  if (!checkIfHaveMoves()) {
+    isPlaying = false;
+    messageLoseElement.classList.remove('hidden');
+  };
+
+  const currentCellsValues = cells.map(cell => cell.textContent);
+
+  if (!isPlaying) {
+    return;
+  }
+
+  switch (e.key) {
+    case 'ArrowDown':
+      moveDirectionAndSum(downDirection, 'down');
+      break;
+    case 'ArrowUp':
+      moveDirectionAndSum(upDirection, 'up');
+      break;
+    case 'ArrowLeft':
+      moveDirectionAndSum(leftDirection, 'left');
+      break;
+    case 'ArrowRight':
+      moveDirectionAndSum(rightDirection, 'right');
+      break;
+  };
+
+  if (!isPlaying) {
+    return;
+  }
```

Comment:
Previous comment about this check, why do you need it? isPlaying can't be changed on lines 210 - 225

Reply:
why it doesn't work if I made changes as of requested?

Reply:
it fixed now with [...], check-up, please